core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -974,20 +1085,29 @@ public GHResponse route( GHRequest request )
             sw = new StopWatch().start();
             Path path = algo.calcPath(fromQResult.getClosestNode(), toQResult.getClosestNode());
             if (path.getTime() < 0)
+            {
                 throw new RuntimeException("Time was negative. Please report as bug and include:" + request);
+            }
 
             paths.add(path);
             debug += ", " + algo.getName() + "-routing:" + sw.stop().getSeconds() + "s, " + path.getDebugInfo();
 
+            // reset all direction enforcements in queryGraph to avoid influencing next path
+            queryGraph.clearUnfavoredStatus();
```

Comment:
ah, we can't move this method into enforceHeading because we potentially call enforceHeading for start and end point, right?